include/vast/CodeGen/CodeGenStmtVisitor.hpp

```diff
@@ -978,8 +978,14 @@ namespace vast::cg {
         operation VisitIndirectCall(const clang::CallExpr *expr) {
             auto callee = VisitIndirectCallee(expr->getCallee())->getResult(0);
             auto args   = VisitArguments(expr);
-            auto type   = hl::getFunctionType(callee.getType(), context().mod.get()).getResults();
-            return make< hl::IndirectCallOp >(meta_location(expr), type, callee, args);
+            auto type   = hl::getFunctionType(callee.getType(), context().mod.get());
+            if (type) {
+                return make< hl::IndirectCallOp >(
+                    meta_location(expr), type.getResults(), callee, args
+                );
+            }
+
+            return {};
```

Comment:
@PappasBrent @xlauko Under what situations is getting a function type here not possible? Can it be documented here in the code with a comment? What about getting the type from `expr->getCallee()`?

Should there be a TODO/FIXME comment?

Reply:
Is this a C++-ism, e.g. calling an `operator()`, or is it related to the issue of builtins? E.g. how Clang has a `BuiltinFn` type (which is super unhelpful and annoying).

Reply:
Thats literally what was there as `VAST_UNIMPEMENTED` and what @PappasBrent was changing in this PR. I don't know every case when this fails but now this is deferred to visitor to yield unsupported type or an error message.

Reply:
I mean a comment describing the case that was hit, so a reader of the code isn't surprised by the fact that one can't always get a function type. I think the code should document the "surprises" of the unhappy path when possible.